### PR TITLE
Make multitest env start/stop failure more explicit for xfail.

### PR DESCRIPTION
**Pull request by yuxuan-ms**
## Bug / Requirement Description
Clearly and concisely describe the problem.

## Solution description
Describe your code changes in detail for reviewers.

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [ ] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: tests/functional/testplan/testing/multitest/test_pre_post_steps.py [hunk 1]
```diff
@@ -61,25 +61,24 @@ def test_pre_post_steps(mockplan):
...
                     TestGroupReport(
-                        name="Before Start",
-                        category=ReportCategories.SYNTHESIZED,
+                        name="Environment Start",
+                        category="synthesized",
                         entries=[
                             TestCaseReport(
                                 name="check_func_1",
-                                category=ReportCategories.SYNTHESIZED,
+                                uid="check_func_1",
```
Comment (zhenyu-ms) [inline tests/functional/testplan/testing/multitest/test_pre_post_steps.py:69]: i think we'd better add "category" back

File: testplan/report/testing/base.py [hunk 2]
```diff
@@ -367,6 +367,9 @@ def propagate_tag_indices(self, parent_tags=None):
         tags_index = tagging.merge_tag_dicts(self.tags, parent_tags or {})
 
         for child in self:
+            # ignore synthesized report
+            if child.category == ReportCategories.SYNTHESIZED:
```
Comment (zhenyu-ms) [inline testplan/report/testing/base.py:371]: will this affect filtering by tags?

File: testplan/testing/base.py [hunk 3]
```diff
@@ -441,34 +445,110 @@ def add_pre_resource_steps(self) -> None:
...
+            case_result.log(
+                f"{driver.__class__.__name__}[{uid}] Status: {driver.status.tag}"
+            )
+
+        case_report.extend(case_result.serialized_entries)
+        case_report.attachments.extend(case_result.attachments)
+        case_report.pass_if_empty()
+        return case_report
+
+    def _add_resource_report(self, name):
```
Comment (zhenyu-ms) [inline testplan/testing/base.py:499]: if no resources && no start/stop related hook, skip synthesize this suite report?

File: testplan/testing/base.py [hunk 4]
```diff
@@ -555,9 +635,20 @@ def _run_error_handler(self) -> None:
...
+            if self.result.report.has_uid(ResourceHooks.error_handler):
+                self.result.report[ResourceHooks.error_handler] = suite_report
+            else:
+                self.result.report.append(suite_report)
+            self._run_resource_hook(
+                self.cfg.error_handler, ResourceHooks.error_handler
+            )
 
-    def _run_resource_hook(self, hook: Callable, label: str) -> None:
+    def _run_resource_hook(self, hook: Callable, report_name: str) -> None:
```
Comment (Pyifan) [inline testplan/testing/base.py:651]: suggest rename parameter report_name --> suite_name

File: testplan/testing/base.py [hunk 5]
```diff
@@ -441,34 +445,110 @@ def add_pre_resource_steps(self) -> None:
...
+            self.result.report[name] = suite_report
+        else:
+            self.result.report.append(suite_report)
+
+    def _finish_resource_report(self, name):
+        self.result.report[name].runtime_status = RuntimeStatus.FINISHED
+
     def add_start_resource_steps(self) -> None:
+        self._add_step(
+            self._add_resource_report, name=ResourceHooks.environment_start
```
Comment (Pyifan) [inline testplan/testing/base.py:514]: suggest that we add environment_start/stop testsuite in _run_resource_hook when not present --> then we have less steps and helps readability

File: testplan/testing/base.py [hunk 6]
```diff
@@ -441,34 +445,110 @@ def add_pre_resource_steps(self) -> None:
...
+    def _dry_run_start_resource(self) -> TestCaseReport:
+        suite_report = self.result.report[ResourceHooks.environment_start]
+        case_report = TestCaseReport(
+            name=ResourceHooks.starting,
+            category=ReportCategories.SYNTHESIZED,
+        )
+        suite_report.append(case_report)
+        return case_report
+
+    def _start_resource(self) -> TestCaseReport:
```
Comment (Pyifan) [inline testplan/testing/base.py:457]: can we do a

if not self.resources: return

here?

File: testplan/testing/base.py [hunk 7]
```diff
@@ -441,34 +445,110 @@ def add_pre_resource_steps(self) -> None:
         self._add_step(self._build_environment)
         self._add_step(self._set_dependencies)
 
+    def _dry_run_start_resource(self) -> TestCaseReport:
```
Comment (Pyifan) [inline testplan/testing/base.py:448]: same here, if self.resources is empty, return

**Review by Pyifan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**